test/integration/scratch-tests.js

```diff
@@ -0,0 +1,113 @@
+/* global vm, Promise */
+const {Chromeless} = require('chromeless');
+const test = require('tap').test;
+const path = require('path');
+const fs = require('fs');
+const chromeless = new Chromeless();
+
+const indexHTML = path.resolve(__dirname, 'index.html');
+const testDir = (...args) => path.resolve(__dirname, 'scratch-tests', ...args);
+
+const testFile = file => test(file, async t => {
+    // start each test by going to the index.html, and loading the scratch file
+    const says = await chromeless.goto(`file://${indexHTML}`)
+        .setFileInput('#file', testDir(file))
+        // the index.html handler for file input will add a #loaded element when it
+        // finishes.
+        .wait('#loaded')
+        .evaluate(() => {
+            // This function is run INSIDE the integration chrome browser via some
+            // injection and .toString() magic.  We can return some "simple data"
+            // back across as a promise, so we will just log all the says that happen
+            // for parsing after.
+
+            // this becomes the `says` in the outer scope
+            const messages = [];
+            const TIMEOUT = 5000;
+
+            vm.runtime.on('SAY', (_, __, message) => {
```

Comment:
What if in the index.html these were stored in a `<pre><code>` wrapper? After the threads complete, we'd then read out that element.

Reply:
I think it's a bit specific - I had imagined we might make other types of integration tests using the same index html/pattern..

I.E. a "click to pick" is a totally different valid integration test here if we could figure out how to script it.

Reply:
I'm personally not very picky about the details of the message collection approach. I always imagined that we'd implement a specific block to emit testing data, but seeing this I'm not sure what the advantage would be if we're crafting the Scratch project specifically for the test.

That said, it might be helpful to have a `<pre><code>` display for the purposes of developing & debugging tests.

Reply:
I don't want to refactor it quite yet for say logging in the index.html yet -- we could add all sorts of debugging tools there I would think, but trying to keep it as simple as I can to ensure the test isn't flaky because of environment.  In other words, let's land without changing this.